src/Movim/i18n/Locale.php

```diff
@@ -179,43 +187,81 @@ public function translate(string $key, $args = false): string
+
+        if (preg_match_all($rexp, $languages, $langParse, PREG_SET_ORDER)) {
+            for ($i = count($langParse) - 1; $i >= 0; $i--) {
+                $locale = locale_parse($langParse[$i]['locale']);
+                if (empty($locale)) {
+                    array_splice($langParse, $i, 1);
+                } else {
+                    $langParse[$i]['locale'] = $locale;
+                    $qual = $langParse[$i]['quality'] ?? null;
+                    $langParse[$i]['quality'] = is_string($qual) ? floatval($qual) : 1.0;
+                }
+            }
+
+            usort($langParse, function($a, $b) {
+                return $a['quality'] - $b['quality'];
+            });
+
+            foreach($langParse as $lp) {
+                $locs[] = $lp['locale'];
+            };
 
-        if (count($lang_parse[1])) {
-            $langs = array_combine($lang_parse[1], $lang_parse[4]);
+            $poFileExists = function (array $l): ?array {
+                $lang = strtolower(implode('_', array_values($l)));
+                return [$lang, file_exists(LOCALES_PATH . $lang . '.po')];
+            };
 
-            foreach ($langs as $lang => $val) {
-                if ($val === '') {
-                    $langs[$lang] = 1;
+            foreach ($locs as $loc) {
+                [$lang, $exists] = $poFileExists($loc);
+                if ($exists) {
+                    $this->language = $lang;
+                    break;
                 }
-            }
-            arsort($langs, SORT_NUMERIC);
-        }
 
-        foreach ($langs as $key => $value) {
-            if (file_exists(LOCALES_PATH . $key . '.po')) {
-                $this->language = $key;
-                break;
-            }
+                if (isset($loc['variant0'])) {
+                    unset($loc['variant0']);
+                    [$lang, $exists] = $poFileExists($loc);
+                    if ($exists) {
+                        $this->language = $lang;
+                        break;
+                    }
+                }
 
-            $exploded = explode('-', $key);
-            $key = reset($exploded);
+                if (isset($loc['script'])) {
+                    unset($loc['script']);
+                    [$lang, $exists] = $poFileExists($loc);
+                    if ($exists) {
+                        $this->language = $lang;
+                        break;
+                    }
+                }
 
-            if (file_exists(LOCALES_PATH . $key . '.po')) {
-                $this->language = $key;
-                break;
-            }
+                if (isset($loc['region'])) {
+                    unset($loc['region']);
+                    [$lang, $exists] = $poFileExists($loc);
+                    if ($exists) {
+                        $this->language = $lang;
+                        break;
+                    }
+                }
 
+                $this->language = 'en';
```

Comment:
Maybe you can just initiate the variable to `en` at the beginning, it will also prevent the next else condition.